facebook/OpenBIC
# fby3.5: bb: Support inform sled cycle

fby3.5: bb: Support inform sled cycle
Summary:
- Add an SEL to inform peer BMC that BB BIC will execute the power sled cycle.

Dependency: #279 

Test plan:
- Build code: Pass
- Inform peer BMC: Pass

Log:
1. Sled cycle on slot1.
- Check log on slot3
root@bmc-oob:~# bic-util slot1 0xE0 0x02 0x9C 0x9C 0x00 0x10 0xC0 0xF0
9C 9C 00 10 31 F0 00 03

root@bmc-oob:~# log-util --print all
1    slot1    2018-03-09 04:36:24    ipmid            SEL Entry: FRU: 1, Record: Standard (0x02), Time: 2018-03-09 04:36:24, Sensor: POWER_DETECT (0xE1), Event Data: (000000) SLED_CYCLE by BB BIC Assertion
1    slot1    2018-03-09 04:35:23    power-util       SERVER_POWER_ON successful for FRU: 1
0    all      2018-03-09 04:35:23    show_sys_config  Abnormal - slot3 instead of slot1
6    nic      2018-03-09 04:35:24    ncsid            FRU: 6 NIC AEN Supported: 0x7, AEN Enable Mask=0x7
6    nic      2018-03-09 04:35:24    ncsid            FRU: 6 PLDM type supported = 0x35
6    nic      2018-03-09 04:35:24    ncsid            FRU: 6 PLDM type 0 version = 1.0.0.0
6    nic      2018-03-09 04:35:24    ncsid            FRU: 6 PLDM type 2 version = 1.1.0.0
6    nic      2018-03-09 04:35:24    ncsid            FRU: 6 PLDM type 4 version = 1.0.0.0
6    nic      2018-03-09 04:35:24    ncsid            FRU: 6 PLDM type 5 version = 1.0.0.0
6    nic      2018-03-09 04:35:24    ncsid            FRU: 6 PLDM sensor monitoring enabled

2. Sled cycle on slot3.
- Check log on slot1
root@bmc-oob:~# bic-util slot1 0xE0 0x02 0x9C 0x9C 0x00 0x10 0xC0 0xF0
9C 9C 00 10 31 F0 00 01
root@bmc-oob:~# log-util --print all
2018 Mar 09 05:05:52 log-util: User cleared all logs
1    slot1    2018-03-09 05:08:13    ipmid            SEL Entry: FRU: 1, Record: Standard (0x02), Time: 2018-03-09 05:08:13, Sensor: POWER_DETECT (0xE1), Event Data: (000000) SLED_CYCLE by BB BIC Assertion
1    slot1    2018-03-09 04:35:23    power-util       SERVER_POWER_ON successful for FRU: 1
6    nic      2018-03-09 04:35:24    ncsid            FRU: 6 NIC AEN Supported: 0x7, AEN Enable Mask=0x7
6    nic      2018-03-09 04:35:24    ncsid            FRU: 6 PLDM type supported = 0x35
6    nic      2018-03-09 04:35:24    ncsid            FRU: 6 PLDM type 0 version = 1.0.0.0
6    nic      2018-03-09 04:35:24    ncsid            FRU: 6 PLDM type 2 version = 1.1.0.0
6    nic      2018-03-09 04:35:24    ncsid            FRU: 6 PLDM type 4 version = 1.0.0.0
6    nic      2018-03-09 04:35:24    ncsid            FRU: 6 PLDM type 5 version = 1.0.0.0
6    nic      2018-03-09 04:35:24    ncsid            FRU: 6 PLDM sensor monitoring enabled
